kili/helpers.py

```diff
@@ -332,3 +333,25 @@ def infer_id_from_external_id(kili, asset_id: str, external_id: str, project_id:
             f'Several assets found containing external ID "{external_id}":'
             f' {assets}. Please, use asset ID instead.')
     return assets[0]['id']
+
+
+def parse_category_search_query(query):
+    """Parse the category search query to assess its validity
+    Args:
+        query: the query to parse
+    """
+    operator = pp.oneOf(">= <= > < ==")
+    number = pp.pyparsing_common.number()
+    dot = '.'
+    word = pp.Word(pp.alphas, pp.alphanums + "_-*")
+    identifier = word + dot + word + dot + 'count'
+    condition = identifier + operator + number
+
+    expr = pp.infixNotation(condition, [
+        ("AND", 2, pp.opAssoc.LEFT, ),
+        ("OR", 2, pp.opAssoc.LEFT, ),
+    ])
+    try:
+        expr.parseString(query, parseAll=True)
+    except pp.ParseException as error:
+        raise ValueError(f'Invalid category search query: {query}') from error
```

Comment:
you should return the error message of pp

Reply:
This is what the `from error`is for. 
Example with: `kili.assets(project_id=project_id, label_category_search = "JOB_0.CLASS_B")`

```---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in _parseNoCache(self, instring, loc, doActions, callPreParse)
   1682                 try:
-> 1683                     loc, tokens = self.parseImpl(instring, preloc, doActions)
   1684                 except IndexError:

~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in parseImpl(self, instring, loc, doActions)
   2896     def parseImpl(self, instring, loc, doActions=True):
-> 2897         if instring[loc] == self.firstMatchChar:
   2898             return loc + 1, self.match

IndexError: string index out of range

During handling of the above exception, another exception occurred:

ParseException                            Traceback (most recent call last)
~/kili-python-sdk/kili/helpers.py in parse_category_search_query(query)
    354     try:
--> 355         expr.parseString(query, parseAll=True)
    356     except pp.ParseException as error:

~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in parseString(self, instring, parseAll)
   1954                     exc.__traceback__ = self._trim_traceback(exc.__traceback__)
-> 1955                 raise exc
   1956         else:

~/anaconda3/envs/python3/lib/python3.9/site-packages/pyparsing.py in _parseNoCache(self, instring, loc, doActions, callPreParse)
   1684                 except IndexError:
-> 1685                     raise ParseException(instring, len(instring), self.errmsg, self)
   1686             else:

ParseException: Expected {OR term | AND term}, found end of text  (at char 13), (line:1, col:14)

The above exception was the direct cause of the following exception:

ValueError                                Traceback (most recent call last)
<ipython-input-3-0c57c8dd9e9c> in <module>
----> 1 kili.assets(project_id='cl3d45bzb0rf70mx4580m0bt1', label_category_search = "JOB_0.CLASS_B")

~/kili-python-sdk/kili/helpers.py in checked_resolver(*args, **kwargs)
     53                     f' {resolver.__name__} with arguments {args}') from exception
     54             if self.client_is_compatible(client_endpoint):
---> 55                 return resolver(*args, **kwargs)
     56             raise EndpointCompatibilityError(
     57                 resolver.__name__, client_endpoint)

~/anaconda3/envs/python3/lib/python3.9/site-packages/typeguard/__init__.py in wrapper(*args, **kwargs)
    808         memo = _CallMemo(python_func, _localns, args=args, kwargs=kwargs)
    809         check_argument_types(memo)
--> 810         retval = func(*args, **kwargs)
    811         check_return_type(retval, memo)
    812 

~/kili-python-sdk/kili/helpers.py in wrapper(*args, **kwargs)
    251             if msg:
    252                 warnings.warn(msg, _type, stacklevel=2)
--> 253             return func(*args, **kwargs)
    254         return wrapper
    255     return decorator

~/kili-python-sdk/kili/queries/asset/__init__.py in assets(self, asset_id, project_id, skip, fields, asset_id_in, consensus_mark_gt, consensus_mark_lt, disable_tqdm, external_id_contains, first, format, honeypot_mark_gt, honeypot_mark_lt, label_author_in, label_consensus_mark_gt, label_consensus_mark_lt, label_created_at, label_created_at_gt, label_created_at_lt, label_honeypot_mark_gt, label_honeypot_mark_lt, label_json_response_contains, label_type_in, metadata_where, skipped, status_in, updated_at_gte, updated_at_lte, as_generator, label_category_search)
    164         disable_tqdm = disable_tqdm or as_generator
    165         if label_category_search:
--> 166             parse_category_search_query(label_category_search)
    167 
    168         payload_query = {

~/kili-python-sdk/kili/helpers.py in parse_category_search_query(query)
    355         expr.parseString(query, parseAll=True)
    356     except pp.ParseException as error:
--> 357         raise ValueError(f'Invalid category search query: {query}') from error

ValueError: Invalid category search query: JOB_0.CLASS_B```

Reply:
@mDuval1